### PR TITLE
[wasm] Log as warning in interop generator when method signature is not supported

### PR DESCRIPTION
**Pull request by maraf**
Currently it only logs a message with low priority. 
It should log warning when pinvoke/icall has a signature that is not supported.

### CODE REVIEW

File: src/tasks/WasmAppBuilder/IcallTableGenerator.cs [hunk 1]
```diff
@@ -178,6 +196,8 @@ private void ProcessType(Type type)
         foreach (var nestedType in type.GetNestedTypes())
             ProcessType(nestedType);
 
+        return !hasError;
```
Comment (radical) [inline src/tasks/WasmAppBuilder/IcallTableGenerator.cs:199]: You can use `Log.HasLoggedErrors` .

File: src/tasks/WasmAppBuilder/IcallTableGenerator.cs [hunk 2]
```diff
@@ -32,21 +32,32 @@ internal sealed class IcallTableGenerator
...
+    public bool TryGenIcallTable(string? runtimeIcallTableFile, string[] assemblies, string? outputPath, [NotNullWhen(true)] out IEnumerable<string>? cookies)
     {
         _icalls.Clear();
         _signatures.Clear();
 
         if (runtimeIcallTableFile != null)
             ReadTable(runtimeIcallTableFile);
 
+        bool hasError = false;
+
```
Comment (radical) [inline src/tasks/WasmAppBuilder/IcallTableGenerator.cs:44]:
```suggestion

```

File: src/tasks/WasmAppBuilder/PInvokeTableGenerator.cs [hunk 3]
```diff
@@ -88,7 +103,7 @@ void CollectPInvokesForMethod(MethodInfo method)
                 string? signature = SignatureMapper.MethodToSignature(method);
                 if (signature == null)
                 {
-                    throw new LogAsErrorException($"Unsupported parameter type in method '{type.FullName}.{method.Name}'");
+                    throw new NotSupportedException($"Unsupported parameter type in method '{type.FullName}.{method.Name}'");
```
Comment (radical) [inline src/tasks/WasmAppBuilder/PInvokeTableGenerator.cs:106]: Why the change? This would show up as an exception with the trace, instead of a neat error message. The trace won't give any extra information, AFAICS.

Comment (maraf) [reply src/tasks/WasmAppBuilder/PInvokeTableGenerator.cs:106]: The exception is caught above. This it gets the same wrapper text as other exceptions.

**Comment by maraf** [comment]
> I think if we want to make this more visible, then we should make it configurable via a msbuild property to either generate an Error, or a Warning. And default to Warning. Users might be using libraries which have signatures with these issues, and they wouldn't be able to get around if we only generate errors for this.

I see. I realized that after changing the time zone unit test. I'll update it to produce warnings be default.

**Comment by lewing** [comment]
this is a tricky one, dllimport can fail at runtime on desktop and there might not be appropriate platform guards on definitions

**Comment by maraf** [comment]
Failures are not relevant

**Comment by radical** [comment]
Maybe it's time to start using some wasm build error codes, like `WS0001` for this particular warning. This would allow users to have `<MSBuildWarningsAsMessages>$(MSBuildWarningsAsMessage);WS0001</MSBuildWarningsAsMessages>` in their project file, and msbuild would convert the warning to a message.

And for this you would need to:

1. provide the `warningCode` to [Log.LogWarning](https://docs.microsoft.com/en-us/dotnet/api/microsoft.build.utilities.tasklogginghelper.logwarning?view=msbuild-17-netcore#microsoft-build-utilities-tasklogginghelper-logwarning(system-string-system-string-system-string-system-string-system-int32-system-int32-system-int32-system-int32-system-string-system-object())).

2. add a `Strings.resx`, and use the resource manager to get the message string from that, to use with `LogWarning`.

This can be done in a follow up PR, and then we can drop the new task parameter, and property.

**Comment by radical** [comment]
> 2. add a `Strings.resx`, and use the resource manager to get the message string from that, to use with `LogWarning`.

you could probably skip this for now.

**Review by radical — APPROVED** [review]
Except for `WasmUnsupportedInteropSignatureAsWarning` -> warning code change, LGTM 👍
